src/app/config-page/config-page.component.spec.ts

```diff
@@ -139,9 +140,69 @@ describe('ConfigPageComponent', () => {
     fixture = TestBed.createComponent(ConfigPageComponent);
     component = fixture.componentInstance;
     fixture.detectChanges();
+    configServiceStub = TestBed.get(ConfigService);
   });
 
   it('should create', () => {
     expect(component).toBeTruthy();
   });
+
+  it('should not save config changes if save is not pressed', done => {
+    fixture.detectChanges();
+    expect(component.displayedButtons).not.toBe(configServiceStub.displayedButtons);
```

Comment:
It would be nice to have some little tests that just check this equality for all the others as well

Reply:
Can't do the same for the primitives.  Have added a test for the others